mcodingbot/plugins/peps.py

```diff
@@ -1,14 +1,23 @@
+from __future__ import annotations
+
 import re
+from contextlib import suppress
+from datetime import datetime, timedelta, timezone
 
 import crescent
 import hikari
+from cachetools import TTLCache
 from crescent.ext import tasks
 
 from mcodingbot.config import CONFIG
 from mcodingbot.utils import PEPManager, Plugin
 
 plugin = Plugin()
 pep_manager = PEPManager()
+MAX_AGE_FOR_SEND = timedelta(minutes=1)
+recent_pep_responses: TTLCache[int, int] = TTLCache(
+    maxsize=100, ttl=MAX_AGE_FOR_SEND.total_seconds() + 5
```

Comment:
The 5's job will be more clear if you do `MAX_AGE_FOR_SEND-5` within the `within_age_cutoff` function

Reply:
I have to disagree. The MAX_AGE_FOR_SEND is exactly how old the message is allowed to be for the bot to try and reply to it. MAX_AGE_FOR_SEND-5 is 5 seconds less than that.

It would be like doing
```py
LOOP_DELAY

while True:
    sleep(LOOP_DELAY-5)
```

Reply:
I see the `within_cutoff` function is asking "is this a safe time to send the message" so it should do adjustments to ensure that there

Reply:
Yeah, doing 5 seconds less doesn't make sense.